packages/lib-classifier/src/store/SubjectViewerStore/SubjectViewerStore.js

```diff
@@ -47,43 +46,28 @@ const SubjectViewer = types
       }
       return false
     },
-    
+ 
+    get hasAnnotateTask () {
+      const canAnnotate = getRoot(self)?.workflowSteps.hasAnnotateTask
+
+      // Make sure the right button is active in the ImageToolbar
+      if (canAnnotate) {
+        self.enableAnnotate()
+      } else {
+        self.enableMove()
+      }
```

Comment:
It's very bad practice to use actions inside a getter like this. Views should not change state, just return derived state that can then be observed/reacted to.

Reply:
As an alternative to `autorun`, you could also handle side effects in your React components, ie. pass `hasAnnotateTask` into the image toolbar as an observable prop, and react to changes in React. 

https://mobx.js.org/the-gist-of-mobx.html#3-create-derivations-that-automatically-respond-to-state-changes

Reply:
I've changed the code to incorporate this approach.